scalalib/src/GenIdeaImpl.scala

```diff
@@ -77,7 +77,7 @@ object GenIdeaImpl {
       else sys.props.get("MILL_BUILD_LIBRARIES") match {
         case Some(found) => found.split(',').map(os.Path(_)).distinct.toList
         case None =>
-          val repos = modules.foldLeft(Set.empty[Repository]) { _ ++ _._2.repositories }
+          val repos = Seq(LocalRepositories.ivy2Local, Repositories.central)
```

Comment:
As far as I understood, there are only two cases where this path is taken: 
* when invoking a mill release
* when invoking the result of `ci/publish-local.sh`

In the first case, I think mill should be fetched from maven central. On the second case, mill is in the local ivy repo.

But on the other hand, this would avoid any kind of private mirror repo, so I think is better to fix it like you suggested